test/execution/server_test.exs

```diff
@@ -142,5 +142,30 @@ defmodule Wanda.Execution.ServerTest do
 
       assert_receive {:DOWN, ^ref, _, ^pid, :normal}
     end
+
+    test "should go down when the timeout function gets called" do
+      execution_id = UUID.uuid4()
+      group_id = UUID.uuid4()
+
+      targets = build_list(3, :target, %{checks: ["expect_check"]})
+
+      {:ok, pid} =
+        start_supervised(
+          {Server,
+           [
+             execution_id: execution_id,
+             group_id: group_id,
+             targets: targets,
+             checks: [Catalog.get_check("expect_check")],
+             config: [timeout: 99_999]
+           ]}
+        )
+
+      ref = Process.monitor(pid)
+
+      Process.send(pid, :timeout, [:noconnect])
+
+      assert_receive {:DOWN, ^ref, _, ^pid, :normal}
+    end
```

Comment:
a test that proves that the timeout is invoked correctly after the timeout period is missing